pandas/core/generic.py

```diff
@@ -5906,6 +5906,95 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def as_nullable_dtypes(
+        self: FrameOrSeries, keep_integer: bool_t = False
+    ) -> FrameOrSeries:
+        """
+        Convert columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        If the dtype is "object", convert to "string", "boolean" or an appropriate
+        integer type.
+
+        If the dtype is "integer", convert to an appropriate integer type.
+
+        If the dtype is numeric, and consists of all integers, convert to an
+        appropriate type.
+
+        Parameters
+        ----------
+        keep_integer : bool, default False
+            Whether ``int`` types should be converted to integer extension types
+
+        Returns
+        -------
+        converted : a copy of the same type as caller
```

Comment:
I don't think this is the form that we typicall do for Returns

Reply:
I'll change it to `same type as input object`, which is what `infer_objects()` uses.